<!-- -*- mode: XML -*- -->

<launch>
  <!-- Node parameters -->

  <!-- Input source -->
  <!-- 1: LidarDet: /LidarDetection -->
  <!-- 2: LidarDet_PointPillars_Car: /LidarDetection/Car -->
  <!-- 3: LidarDet_PointPillars_Ped_Cyc: /LidarDetection/Ped_Cyc -->
  <!-- 4: VirtualBBoxAbs: /abs_virBB_array -->
  <!-- 5: VirtualBBoxRel: /rel_virBB_array -->
  <!-- 6: CameraDetV2: /CameraDetection -->
  <!-- 7: Tracking2D: /Tracking2D/front_bottom_60 -->
  <!-- otherwise: Fusion: /SensorFusion -->
  <arg name="input_source" default="1" />

  <!-- Occupancy source -->
  <!-- 1: MapBased: /occupancy_wayarea -->
  <!-- otherwise: PlannedPathBased: /occupancy_grid_wayarea -->
  <arg name="occ_source" default="0" />

  <!-- Save output to .txt file-->
  <arg name="save_output_txt" default="False" />

  <!-- Output bbox and pp points in tf_map -->
  <arg name="output_tf_map" default="False" />

  <!-- Input/Output FPS -->
  <arg name="input_fps" default="10.0" />
  <arg name="output_fps" default="10.0" />

  <!-- Create bbox(bPoint) from polygon(cPoint) -->
  <arg name="create_bbox_from_polygon" default="False" />

  <!-- Create polygon(cPoint) from bbox(bPoint) -->
  <arg name="create_polygon_from_bbox" default="False" />

  <!-- Move PP AR(1) input's origin to (pp_input_shift, pp_input_shift) -->
  <arg name="pp_input_shift_m" default="150.0" />

  <!-- Rviz marker lifetime -->
  <!-- 0: Auto-assign as (1 / output_fps) -->
  <!-- other value: manually assign -->
  <arg name="m_lifetime_sec" default="0.15" />

  <!-- Rviz markers -->
  <arg name="gen_markers" default="True" />
  <arg name="show_classid" default="False" />
  <arg name="show_tracktime" default="False" />
  <arg name="show_source" default="False" />
  <arg name="show_distance" default="False" />
  <arg name="show_absspeed" default="True" />

  <!-- Min number of consecutive tracked time ticks to activate PP: should be >= 3 -->
  <arg name="num_pp_input_min" default="4" />

  <arg name="pp_obj_min_kmph" default="1.0" />
  <arg name="pp_obj_max_kmph" default="80.0" />

  <!-- Launch node -->
  <node pkg="itri_tracking_pp" type="itri_tracking_pp" name="itri_tracking_pp" cwd="node" output="screen">
    <param name="input_source" value="$(arg input_source)" />
    <param name="occ_source" value="$(arg occ_source)" />
    <param name="save_output_txt" value="$(arg save_output_txt)" />
    <param name="output_tf_map" value="$(arg output_tf_map)" />
    <param name="input_fps" value="$(arg input_fps)" />
    <param name="output_fps" value="$(arg output_fps)" />
    <param name="create_bbox_from_polygon" value="$(arg create_bbox_from_polygon)" />
    <param name="create_polygon_from_bbox" value="$(arg create_polygon_from_bbox)" />
    <param name="pp_input_shift_m" value="$(arg pp_input_shift_m)" />
    <param name="m_lifetime_sec" value="$(arg m_lifetime_sec)" />
    <param name="gen_markers" value="$(arg gen_markers)" />
    <param name="show_classid" value="$(arg show_classid)" />
    <param name="show_tracktime" value="$(arg show_tracktime)" />
    <param name="show_source" value="$(arg show_source)" />
    <param name="show_distance" value="$(arg show_distance)" />
    <param name="show_absspeed" value="$(arg show_absspeed)" />
    <param name="num_pp_input_min" value="$(arg num_pp_input_min)" />
    <param name="pp_obj_min_kmph" value="$(arg pp_obj_min_kmph)" />
    <param name="pp_obj_max_kmph" value="$(arg pp_obj_max_kmph)" />
  </node>
</launch>
